### PR TITLE
Initial reorg

### PR DESCRIPTION
**Pull request by ohler55**
A first attempt at reorganizing. Comments and suggestions would be appreciated.

### CODE REVIEW

**Comment by ashmaroli** [comment]
A couple of things not clear outright:
 - Is WAB a single-page JS App running on Ruby API or a Ruby Framework to help build an App?
 - If its the latter, WAB shouldn't be dependent on JS at all.. the JS is simply an implementation detail
 - Another thing not clear is, the Ruby code has three `Shell` classes : `WAB::Shell`, `WAB::Impl::Shell`, `WAB::IO::Shell`. From an end-user point-of-view, what is the significance of these three classes?

File: pages/Architecture.md [hunk 1]
```diff
@@ -1,11 +1,52 @@
 # Architecture
 
+WABuR is a Web Application Builder using Ruby. It strives to keep be simple to
+use and maintain not only for the trivial case but for more advanced
```
Comment (ashmaroli) [inline pages/Architecture.md]: consider rewording this sentence.. does not flow smoothly..

File: pages/Architecture.md [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
+applications as well. It utilizes a Mode/View/Controller pattern with well
+defined APIs between each element of the MVC approach.
+
+One of the more interesting benefits of well defined APIs between MVC
+components is that it allows multiple Views without changing any other
+components. A View can be a single page JavaScript application asis used in
```
Comment (ashmaroli) [inline pages/Architecture.md]: typo
```diff
- asis used
+ as used
```

File: pages/Architecture.md [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+the WABuR reference implementation or it can simply be `curl` used to send
```
Comment (ashmaroli) [inline pages/Architecture.md]:
```diff
- simply be `curl` used to send.. as is used
+ simply be  a `curl` command used to send.. as used
```

**Comment by ashmaroli** [comment]
should `sass` be listed as a dependency in the gemspec?

**Comment by ohler55** [comment]
I think sass should be listed as a dependency.

**Comment by ohler55** [comment]
Sass should be added but not part of a documentation PR.

**Comment by ashmaroli** [comment]
Agreed..

**Comment by ohler55** [comment]
Have you completed the review?

**Pull request merged**